### PR TITLE
Fix sync problem on epoch change

### PR DESCRIPTION
**Pull request by SebastianMarian**

### CODE REVIEW

File: process/sync/baseSync.go [hunk 1]
```diff
@@ -298,25 +294,46 @@ func (boot *baseBootstrap) ShouldSync() bool {
...
 
 	roundDiff := uint64(boot.rounder.Index()) - currHeader.GetRound()
-	if roundDiff < tryRequestHeaderDelta {
+	if roundDiff <= process.MaxRoundsWithoutNewBlockReceived {
 		return
 	}
 
-	shouldTrySync := roundDiff%tryRequestHeaderDelta == 0
+	numRequestedHdrs := 0
+	nonce := boot.getNonceForNextBlock()
```
Comment (iulianpascalau) [inline process/sync/baseSync.go]: is this concurrent safe?

Comment (SebastianMarian) [reply process/sync/baseSync.go]: If this is not concurrent safe that means we have a problem on each call from chainHandler. This is the method listed:

// getNonceForNextBlock will get the nonce for the next block we should request
func (boot *baseBootstrap) getNonceForNextBlock() uint64 {
	nonce := uint64(1) // first block nonce after genesis block
	if boot.chainHandler != nil && boot.chainHandler.GetCurrentBlockHeader() != nil {
		nonce = boot.chainHandler.GetCurrentBlockHeader().GetNonce() + 1
	}
	return nonce
}

File: process/sync/baseSync.go [hunk 2]
```diff
@@ -298,25 +294,46 @@ func (boot *baseBootstrap) ShouldSync() bool {
 	}
 	boot.statusHandler.SetUInt64Value(core.MetricIsSyncing, result)
 
+	go boot.requestHeadersIfSyncIsStuck()
```
Comment (sasurobert) [inline process/sync/baseSync.go]: no need for go routine.

File: process/sync/baseSync.go (hunk 1 above)
Comment (sasurobert) [reply process/sync/baseSync.go]: check.ifnil

Comment (SebastianMarian) [reply process/sync/baseSync.go]: Refactored

File: process/sync/baseSync.go (hunk 2 above)
Comment (SebastianMarian) [reply process/sync/baseSync.go]: Refactored

**Pull request merged**